src/pages/rekey-search/rekey-search.effects.ts

```diff
@@ -2,33 +2,51 @@ import { Injectable } from '@angular/core';
 import { Effect, Actions, ofType } from '@ngrx/effects';
 import { RekeySearchProvider } from '../../providers/rekey-search/rekey-search';
 import { switchMap, map, catchError } from 'rxjs/operators';
-import * as rekeySearchActions from './rekey-search.actions';
 import { of } from 'rxjs/observable/of';
 import { CompressionProvider } from '../../providers/compression/compression';
+import { SearchProvider } from '../../providers/search/search';
+import {
+  SEARCH_BOOKED_TEST, SearchBookedTest, SearchBookedTestSuccess, SearchBookedTestFailure,
+} from './rekey-search.actions';
+import { RekeySearchErrorMessages } from './rekey-search-error-model';
+import { HttpResponse } from '@angular/common/http';
+import { HttpStatusCodes } from '../../shared/models/http-status-codes';
 
 @Injectable()
 export class RekeySearchEffects {
 
   constructor(
     private actions$: Actions,
+    private testSearchProvider: SearchProvider,
     private rekeySearchProvider: RekeySearchProvider,
     private compressionProvider: CompressionProvider,
   )
   { }
 
   @Effect()
-  getTest$ = this.actions$.pipe(
-    ofType(rekeySearchActions.SEARCH_BOOKED_TEST),
-    switchMap((action: rekeySearchActions.SearchBookedTest) => {
-      const rekeySearchParams = {
-        applicationReference: action.appRef,
-        staffNumber: action.staffNumber,
-      };
-      return this.rekeySearchProvider.getTest(rekeySearchParams).pipe(
-        map(response => this.compressionProvider.extractTestSlotResult(response.toString())),
-        map((testSlot: any) => new rekeySearchActions.SearchBookedTestSuccess(testSlot, action.staffNumber)),
+  getBooking$ = this.actions$.pipe(
+    ofType(SEARCH_BOOKED_TEST),
+    switchMap((action: SearchBookedTest) => {
+      return this.testSearchProvider.getTestResult(action.appRef, action.staffNumber).pipe(
+        switchMap((response: HttpResponse<any>) => {
+          if (response.status === HttpStatusCodes.OK) {
+            return of(new SearchBookedTestFailure({
+              message: RekeySearchErrorMessages.BookingAlreadyCompleted,
+            }));
+          }
+        }),
         catchError((err: any) => {
-          return of(new rekeySearchActions.SearchBookedTestFailure(err));
+          const rekeySearchParams = {
+            applicationReference: action.appRef,
+            staffNumber: action.staffNumber,
+          };
+          return this.rekeySearchProvider.getBooking(rekeySearchParams).pipe(
+            map(response => this.compressionProvider.extractTestSlotResult(response.toString())),
+            map((testSlot: any) => new SearchBookedTestSuccess(testSlot, action.staffNumber)),
```

Comment:
I'm having mixed feelings about this success scenario being dealt with within a `catchError` block :P
Not sure atm if there is a better way to do it though :)

Reply:
I could make it more specific to be fair, looking out for the specific status code that denotes there is no submitted test.

Reply:
But yeah not sure there's a way around the catchError :(

Reply:
Have made some amendments if you'd like to re-review @rstepnowski